clients/python/setup.py

```diff
@@ -31,10 +27,16 @@
 with open('README.rst') as f:
     readme = f.read()
 
+init = os.path.join(os.path.dirname(__file__), 'girder_client', '__init__.py')
+with open(init) as fd:
+    version = re.search(
+        r'^__version__\s*=\s*[\'"]([^\'"]*)[\'"]',
+        fd.read(), re.MULTILINE).group(1)
+
```

Comment:
The problem is that when you `import girder_client`, it [transitively imports a bunch of stuff](https://github.com/girder/girder/blob/master/clients/python/girder_client/__init__.py#L20-L31). This works fine if you are installing via an egg or have already installed the requirements, but if you checked out the girder source and tried to do a `pip install -e .` (which normally works), it would no longer work because none of the required third party libraries would have been installed yet.